specification/network/resource-manager/Microsoft.Network/stable/2024-01-01/virtualNetwork.json

```diff
@@ -1684,6 +1687,10 @@
           "type": "string",
           "description": "The type of the endpoint service."
         },
+        "networkIdentifier": {
+          "type": "string",
+          "description": "A public IP as network identification."
```

Comment:
In your example, it seems to be the "Azure resource ID of a public IP..."

>"networkIdentifier": "/subscriptions/subid/resourceGroups/subnet-test/providers/Microsoft.Network/publicIPAddresses/test-ip",

Please add "format": "arm-id" if thats correct.